### DISCUSSION TITLE
setOrderAddress and active order

### DISCUSSION THREAD

**Discussion by ivasilov** [vendurehq/vendure#1929] [General]
Hi,

We're thinking of changing our store flow by selecting an address first (due to some products not being available in certain shipping zones).

Looking through the code, we saw that `setOrderShippingAddress` and `setOrderBillingAddress` are not creating an order. Shouldn't they be the same as `addItemToOrder`? Or at least, add an optional flag whether to create the order or not.

I can submit a PR if this looks like a good idea.

**Comment by michaelbromley** [comment]
Hi,
Yes, I think it's reasonable to expect those 2 mutations to implicitly create an active order in the same way that e.g. `addItemToOrder` does. 

A PR would be welcome! 👍
